Consul/Configuration.cs

```diff
@@ -0,0 +1,1736 @@
+        public LimitsConfig Limits { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public PassiveHealthCheckConfig PassiveHealthCheck { get; set; }
+    }
+
+    public class LimitsConfig
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public int? MaxConnections { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public int? MaxPendingRequests { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public int? MaxConcurrentRequests { get; set; }
+    }
+
+    public class EnvoyExtensionConfig
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Name { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public bool Required { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public Dictionary<string, string> Arguments { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string ConsulVersion { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string EnvoyVersion { get; set; }
+    }
+
+    public class DestinationConfig
+    {
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public List<string> Addresses { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public int? Port { get; set; }
+    }
+
+    public interface IConfigurationEntry
+    {
+        string Kind { get; set; }
+    }
+
+    public class Configuration : IConfigurationEndpoint
+    {
+        private readonly ConsulClient _client;
+
+        internal Configuration(ConsulClient c)
+        {
+            _client = c;
+        }
+
+        /// <summary>
+        ///  This creates or updates the given config entry.
+        /// </summary>
+        /// <param name="q">Write Options</param>
+        /// <param name="configurationEntry">The configuration entry</param>
+        /// <returns>An empty write result</returns>
+        public Task<WriteResult> ApplyConfig<TConfig>(WriteOptions q, TConfig configurationEntry, CancellationToken ct = default) where TConfig : IConfigurationEntry
+        {
+            var req = _client.Put("/v1/config", configurationEntry, q);
+            return req.Execute(ct);
+        }
+
+        /// <summary>
+        ///  This creates or updates the given config entry.
+        /// </summary>
+        /// <param name="configurationEntry">The configuration entry</param>
+        /// <returns>An empty write result</returns>
+        public Task<WriteResult> ApplyConfig<TConfig>(TConfig configurationEntry, CancellationToken ct = default) where TConfig : IConfigurationEntry
+        {
+            var req = _client.Put("/v1/config", configurationEntry, WriteOptions.Default);
```

Comment:
You can call the other variant of this method from here, so we avoid repeating the code.